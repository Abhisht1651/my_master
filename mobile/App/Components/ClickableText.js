// @flow
import React from 'react'
import { Text, TouchableOpacity, StyleSheet } from 'react-native'
import PropTypes from 'prop-types'

type Props = {
  onPress?: PropTypes.func,
  textStyle?: StyleSheet,
  children?: string,
  style?: StyleSheet,
  text?: string
}
const ClickableText = (props: Props) => {
  const { style, onPress, textStyle, children, text } = props
  return (
    <TouchableOpacity style={style} onPress={onPress}>
      <Text style={textStyle}> {children || text} </Text>
    </TouchableOpacity>
  )
}

ClickableText.defaultProps = {
  onPress: () => {},
  textStyle: StyleSheet.create({}),
  children: '',
  style: StyleSheet.create({}),
  text: ''
}

export default ClickableText
