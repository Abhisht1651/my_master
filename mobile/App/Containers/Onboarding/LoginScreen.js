/* eslint-disable no-unused-vars */
// @flow
import { NavigationContainerProps } from '@react-navigation/native'

import React, { useState } from 'react'
import { Text, Image, View, TouchableOpacity, Platform } from 'react-native'
import { responsiveHeight } from 'react-native-responsive-dimensions'
import { Images } from '../../Themes'

// Styles
import styles from '../Styles/Onboarding/LoginScreenStyles'
import ClickableText from '../../Components/ClickableText'
import InputText from '../../Components/TextInput/TextInputComponent'
import PasswordText from '../../Components/TextInput/TextInputForPassword'
import { MediumButton } from '../../Components/Buttons/MediumButton'

type Props = NavigationContainerProps & {}
const LoginScreen = (props: Props) => {
  const [userInfo, setUserInfo] = useState('')
  const [userPassword, setPassword] = useState('')
  return (
    <View style={styles.mainContainer}>
      {Platform.OS === 'ios' && <View style={styles.statusBar} />}

      <View style={styles.mainContainer}>
        <TouchableOpacity
          onPress={() => props.navigation.goBack()}
          style={styles.backArrowContainer}>
          <Image source={Images.leftArrow} style={styles.backArrow} />
        </TouchableOpacity>

        <View style={styles.section}>
          <Image source={Images.newzeraPurpleLogo} style={styles.logo} />
          <View style={[styles.InputText, { marginTop: responsiveHeight(9.25) }]}>
            <InputText placeholder="Email / Phone No." captureText={(text) => setUserInfo(text)} />
          </View>
          <View style={[styles.InputText, { marginTop: responsiveHeight(3.2) }]}>
            <PasswordText placeholder="Password" captureText={(text) => setPassword(text)} />
          </View>
        </View>
        <View style={styles.buttonWrap}>
          <MediumButton
            title="Continue"
            onPress={() => props.navigation.navigate('WelcomeLogin')}
          />

          <ClickableText
            style={styles.forgotPass}
            textStyle={styles.forgotPassText}
            onPress={() => props.navigation.navigate('ForgotPasswordScreen')}
            text="Forgot password"
          />
          <View style={styles.sectionRow}>
            <Text style={styles.signUpTextMain}>Not an user? </Text>
            <ClickableText
              textStyle={styles.signUpTextSignUp}
              onPress={() => props.navigation.navigate('SignUp')}
              text="Sign up"
            />
          </View>
        </View>
      </View>
    </View>
  )
}

export default LoginScreen
