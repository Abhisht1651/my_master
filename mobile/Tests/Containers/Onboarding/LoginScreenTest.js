import { Image, Text } from 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import LoginScreen from '../../../App/Containers/Onboarding/LoginScreen'
import InputText from '../../../App/Components/TextInput/TextInputComponent'
import PasswordText from '../../../App/Components/TextInput/TextInputForPassword'
import { MediumButton } from '../../../App/Components/Buttons/MediumButton'
import ClickableText from '../../../App/Components/ClickableText'

const createTestProps = (props) => ({
  navigation: {
    navigate: jest.fn()
  },
  ...props
})

describe('LoginScreen', () => {
  describe('rendering', () => {
    let wrapper
    let props
    beforeEach(() => {
      props = createTestProps({})
      wrapper = shallow(<LoginScreen {...props} />) // no compile-time error
    })
    it('should match to snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('should render 2 <Image />', () => {
      expect(wrapper.find(Image)).toHaveLength(2)
    })

    it('should render 1 <InputText />', () => {
      expect(wrapper.find(InputText)).toHaveLength(1)
    })
    it('should render 1 <PasswordText />', () => {
      expect(wrapper.find(PasswordText)).toHaveLength(1)
    })
    it('1st <InputText /> is email', () => {
      const inputText = wrapper.find(InputText).at(0)
      expect(inputText.prop('placeholder')).toBe('Email / Phone No.')
    })
    it('2nd <PasswordText /> is password', () => {
      const inputText = wrapper.find(PasswordText).at(0)
      expect(inputText.prop('placeholder')).toBe('Password')
    })
    it('should render 1 <Text /> tag', () => {
      expect(wrapper.find(Text)).toHaveLength(1)
    })

    it('should render 1 <Rounded Button />', () => {
      expect(wrapper.find(MediumButton)).toHaveLength(1)
    })

    it('should render <ClickableText /> tag', () => {
      expect(wrapper.find(ClickableText)).toHaveLength(2)
    })
    it('should navigate to ForgotPasswordScreen Screen when clicked on Forgot Password Text', () => {
      const button1 = wrapper.find(ClickableText).at(0)
      button1.simulate('press')
      expect(props.navigation.navigate).toHaveBeenCalledWith('ForgotPasswordScreen')
    })
    it('should navigate to SignUp Screen when clicked on SignUp Text', () => {
      const button1 = wrapper.find(ClickableText).at(1)
      button1.simulate('press')
      expect(props.navigation.navigate).toHaveBeenCalledWith('SignUp')
    })
  })
})
