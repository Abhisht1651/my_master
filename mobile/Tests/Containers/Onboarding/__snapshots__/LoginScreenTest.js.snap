// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen rendering should match to snapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "height": 58.188802083333336,
        "width": "100%",
      }
    }
  />
  <View
    style={
      Object {
        "backgroundColor": "white",
        "flex": 1,
      }
    }
  >
    <ForwardRef
      onPress={[Function]}
      style={
        Object {
          "left": 54,
          "position": "absolute",
          "top": 2.60546875,
        }
      }
    >
      <Image
        source={
          Object {
            "default": "",
          }
        }
        style={
          Object {
            "height": 31.265625,
            "resizeMode": "contain",
            "width": 36,
          }
        }
      />
    </ForwardRef>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <Image
        source={
          Object {
            "default": "",
          }
        }
        style={
          Object {
            "height": 216.108,
            "marginTop": 236.25140000000002,
            "resizeMode": "contain",
            "width": 434.99999999999994,
          }
        }
      />
      <View
        style={
          Array [
            Object {
              "width": 549.75,
            },
            Object {
              "marginTop": 123.395,
            },
          ]
        }
      >
        <TextInputComponent
          borderBottomColor="#7B46E4"
          borderBottomStyle={Object {}}
          captureText={[Function]}
          onBlur={[Function]}
          onFocus={[Function]}
          placeholder="Email / Phone No."
          placeholderTextColor="#8F8FAF"
        />
      </View>
      <View
        style={
          Array [
            Object {
              "width": 549.75,
            },
            Object {
              "marginTop": 42.688,
            },
          ]
        }
      >
        <TextInputForPassword
          borderBottomColor="#7B46E4"
          borderBottomStyle={Object {}}
          captureText={[Function]}
          onBlur={[Function]}
          onFocus={[Function]}
          placeholder="Password"
          placeholderTextColor="#8F8FAF"
        />
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
          "marginTop": 53.36,
        }
      }
    >
      <MediumButton
        disabled={false}
        onPress={[Function]}
        outline={false}
        style={Object {}}
        title="Continue"
      />
      <ClickableText
        onPress={[Function]}
        style={
          Object {
            "marginTop": 34.684000000000005,
          }
        }
        text="Forgot password"
        textStyle={
          Object {
            "color": "#7B46E4",
            "fontFamily": "BananaGrotesk-Bold",
            "fontSize": 28,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "center",
            "marginTop": 200.1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#1F1F60",
              "fontFamily": "BananaGrotesk-Regular",
              "fontSize": 31,
            }
          }
        >
          Not an user? 
        </Text>
        <ClickableText
          onPress={[Function]}
          style={Object {}}
          text="Sign up"
          textStyle={
            Object {
              "color": "#7B46E4",
              "fontFamily": "BananaGrotesk-Bold",
            }
          }
        />
      </View>
    </View>
  </View>
</View>
`;
